.travis.yml

```diff
@@ -1,14 +1,26 @@
-language: python
-
-python:
-  - "2.7"
+language: generic
 
 addons:
   postgresql: '9.4'
 
 before_script:
-    - yes | python setup.py install
-    - pip install -r dev-requirements.txt
-    - python bin/destroy_and_setup_psqlgraph.py
-    - pip freeze
-script: "py.test -v"
+  - psql -U postgres -c "create user test with superuser password 'test';"
+  - psql -U postgres -c "create database automated_test with owner test;"
+  - unset PYENV_ROOT
+  - rm -rf ~/.pyenv
+  - curl -L https://github.com/pyenv/pyenv-installer/raw/master/bin/pyenv-installer | bash
+  - export PATH="$HOME/.pyenv/bin:$PATH"
+  - pyenv install 3.5.7
+  - pyenv install 3.6.9
+
+install:
+  - virtualenv venv
+  - source venv/bin/activate
+  - pip install tox-travis tox-pyenv
```

Comment:
Third-party dependencies versions should also be deterministic.  Pinning versions or using lock files are examples of how to manage this.  I'm not familiar enough with Python's ecosystem to have a specific recommendation yet.